tests/unittests/tests-checksum/tests-checksum-crc8.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 Benjamin Valentin <benpicco@googlemail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "checksum/crc8.h"
+
+#include "tests-checksum.h"
+
+#define CRC8_POLY   0x31
+#define CRC8_INIT   0xff
+
+static void test_checksum_crc8_sequence_empty(void)
+{
+    unsigned char buf[] = "";
+    uint8_t expect = 0xFF;
+
+    TEST_ASSERT_EQUAL_INT(expect, crc8(buf, sizeof(buf) - 1, CRC8_POLY, CRC8_INIT));
+}
+
+static void test_checksum_crc8_sequence_1a(void)
+{
+    unsigned char buf[] = "A";
+    uint8_t expect = 0xA0;
+
+    TEST_ASSERT_EQUAL_INT(expect, crc8(buf, sizeof(buf) - 1, CRC8_POLY, CRC8_INIT));
+}
+
+static void test_checksum_crc8_sequence_256a(void)
+{
+    unsigned char buf[] = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
```

Comment:
Maybe it would be better for readability to define the test string in a regular order, e.g. 8 times 32 characters
```suggestion
    unsigned char buf[] = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
```